libkbfs/bserver_remote.go

```diff
@@ -369,17 +369,33 @@ func (b *BlockServerRemote) batchDowngradeReferences(ctx context.Context,
 			if !tmpErr {
 				finalError = err
 				bid, err := BlockIDFromString(res.Failed.Bid.BlockHash)
+				found := false
 				if err == nil {
 					if refs, ok := contexts[bid]; ok {
 						for i := range refs {
 							if refs[i].GetRefNonce() == BlockRefNonce(res.Failed.Nonce) {
 								refs = append(refs[:i], refs[i+1:]...)
 								contexts[bid] = refs
+								// It's possible that the failure
+								// happened due to a complete network
+								// error.  If the zero block ref
```

Comment:
I don't think this is true -- a complete network failure would mean that the `BlockHash` would be the zero value, which would fail in `BlockIDFromString`.